tests/nlu/classifiers/test_diet_classifier.py

```diff
@@ -578,114 +524,58 @@ async def test_train_tensorboard_logging(
 
 
 async def test_train_model_checkpointing(
-    component_builder: ComponentBuilder, tmp_path: Path, nlu_data_path: Text,
+    default_model_storage: ModelStorage,
+    default_diet_resource: Resource,
+    create_train_load_and_process_diet: Callable[..., Message],
 ):
-    model_name = "nlu-checkpointed-model"
-    model_dir = tmp_path / model_name
-    checkpoint_dir = get_checkpoint_dir_path(tmp_path, model_dir)
-    assert not checkpoint_dir.is_dir()
-
-    _config = RasaNLUModelConfig(
-        {
-            "pipeline": [
-                {"name": "WhitespaceTokenizer"},
-                {"name": "CountVectorsFeaturizer"},
-                {
-                    "name": "DIETClassifier",
-                    EPOCHS: 2,
-                    EVAL_NUM_EPOCHS: 1,
-                    EVAL_NUM_EXAMPLES: 10,
-                    CHECKPOINT_MODEL: True,
-                },
-            ],
-            "language": "en",
-        }
+    create_train_load_and_process_diet(
+        {EPOCHS: 2, EVAL_NUM_EPOCHS: 1, EVAL_NUM_EXAMPLES: 10, CHECKPOINT_MODEL: True},
     )
 
-    await rasa.nlu.train.train(
-        _config,
-        path=str(tmp_path),
-        data=nlu_data_path,
-        component_builder=component_builder,
-        fixed_model_name=model_name,
-    )
+    with default_model_storage.read_from(default_diet_resource) as model_dir:
```

Comment:
shouldn't we rather grab the self._resource from the model directly?

Reply:
Could do, although it's a protected member, and we know this will be the same one.

Reply:
As a lazy person I don't like having to look into create_diet here to find out they're the same :)  - but I also agree that doing that via protected variable isn't nice and there isn't really a reason to introduce some getter because the models have the .load already 🤔 . So I'm fine leaving it as is.